src/shared/components/palette/case-flag/components/search-language-interpreter/search-language-interpreter.component.ts

```diff
@@ -0,0 +1,82 @@
+import { ChangeDetectorRef, Component, EventEmitter, Input, OnInit, Output } from '@angular/core';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { Observable } from 'rxjs';
+import { map } from 'rxjs/operators';
+import { ErrorMessage } from '../../../../../domain';
+import { CaseFlagState, Language } from '../../domain';
+import { CaseFlagFieldState } from '../../enums';
+
+@Component({
+  selector: 'ccd-search-language-interpreter',
+  templateUrl: './search-language-interpreter.component.html',
+  styleUrls: ['./search-language-interpreter.component.scss']
+})
+export class SearchLanguageInterpreterComponent implements OnInit {
+
+  @Input()
+  public formGroup: FormGroup;
+
+  @Output()
+  public caseFlagStateEmitter: EventEmitter<CaseFlagState> = new EventEmitter<CaseFlagState>();
+
+  public readonly minSearchCharacters = 3;
+  public languages: Language[];
+  public filteredLanguages$: Observable<string[]>;
+  public showAutocomplete = false;
+  public term = '';
+  public isCheckboxEnabled = false;
+  public errorMessages: ErrorMessage[];
+
+  constructor(private readonly fb: FormBuilder, private readonly cd: ChangeDetectorRef) {
+    this.formGroup = this.fb.group({
+      searchTerm: ['']
+    });
+  }
+
+  public ngOnInit(): void {
+    this.filteredLanguages$ = this.formGroup.controls.searchTerm.valueChanges.pipe(
+      map(value => this.getLanguages(value))
+    );
+  }
+
+  public onNext(): void {
+    // Validate form
+    this.validateForm();
+    // Return case flag field state and error messages to the parent
+    this.caseFlagStateEmitter.emit({
+      currentCaseFlagFieldState: CaseFlagFieldState.FLAG_LANGUAGE_INTERPRETER,
+      errorMessages: this.errorMessages
+    });
+  }
+
+  public onEnterLanguageManually(event: any): void {
```

Comment:
@johnbenjamin-hmcts - please check if there is a suitable alternative to type `any`, for example, `Event`. I'm not sure on this, though... maybe there isn't one.